text/0026-activation-distribution.md

```diff
@@ -0,0 +1,120 @@
+# Project title
+- Start Date: 2020/04/27
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/26
+
+## Decision, including impact on distributions
+OpenMRS Microfrontends will have their own default activator functions.
+Microfrontends will be identified as such by their exports satisfying a
+specific microfrontend interface, defined in this RFC. The app shell will load
+the app shell config if available. The app shell will register all
+microfrontends in the import map with Single SPA, and start the application.
+
+The existing concept of a "root config" will be superseded by the app shell
+and an app shell config.
+
+In order to implement this change, we will need to
+
+1. Add index.ts entrypoint files to each microfrontend that satisfy the
+microfrontend interface (exporting lifecycle and activate).
+2. Change the initial script to behave as described.
+3. Migrate all existing root configs to the new structure. This will generally
+just mean removing the calls to registerApplication and exporting those
+activators that are different from the defaults.
+
+Instead of lazy loading all modules from the importmaps the modules are
+directly evaluated. 
+
+## Definition
+An **activation function** is a JavaScript function that is used by the Single
+SPA library to determine if and when a microfrontend should be activated.
+
+The **activation of a microfrontend** describes when the lifecycle of an app
+should begin.
+
+The **lifecycle** of an app determines what to do technically for booting,
+mounting, updating, and unmounting an application in the DOM of the currently
+presented website.
+
+The **DOM** refers to the API responsible for the object representation of a
+website in the browser.
+
+An **app shell** refers to the index.html page that includes the required base
+scripts. Thus triggering the whole loading process necessary for all the
+microfrontends.
+
+The **initial script** is the script performing the loading of the Single SPA
+library. It sets up important globals and imports the important global modules.
+
+The new **root config** replaces the old concept of a [root config][1]. An
+optional element of the import map, named `@openmrs/esm-root-config`, which can
```

Comment:
```suggestion
optional element of the import map, named `root-config`, which can
```

My understanding is that `@openmrs/esm-root-config` will either cease to exist or be converted into the *initial script*, right? This new "root config" (maybe we should call it something else?) will be totally different. There will be no default version of it, it's totally optional and implementation-specific. So it wouldn't make sense to namespace it to `@openmrs`. Does that make sense?